src/components/macro/macro.component.ts

```diff
@@ -1,31 +1,41 @@
-import { Component, OnDestroy, OnInit, QueryList, ViewChildren } from '@angular/core';
+import { Component, QueryList, ViewChildren } from '@angular/core';
 import { ActivatedRoute } from '@angular/router';
 
-import { Subscription } from 'rxjs/Subscription';
+import { Store } from '@ngrx/store';
+import 'rxjs/add/operator/do';
+import 'rxjs/add/operator/let';
+import 'rxjs/add/operator/publishReplay';
+import 'rxjs/add/operator/switchMap';
+import { Observable } from 'rxjs/Observable';
+import { ConnectableObservable } from 'rxjs/observable/ConnectableObservable';
 
 import { DragulaService } from 'ng2-dragula/ng2-dragula';
 
 import { Macro } from '../../config-serializer/config-items/Macro';
-import { MacroAction } from '../../config-serializer/config-items/macro-action';
+import { MacroAction } from '../../config-serializer/config-items/macro-action/MacroAction';
 import { MacroItemComponent } from './item/macro-item.component';
 
-import { UhkConfigurationService } from '../../services/uhk-configuration.service';
+import { AppState } from '../../store';
+import { MacroActions } from '../../store/actions';
+import { getMacro } from '../../store/reducers/macro';
 
 @Component({
     selector: 'macro',
     template: require('./macro.component.html'),
     styles: [require('./macro.component.scss')],
     viewProviders: [DragulaService]
 })
-export class MacroComponent implements OnInit, OnDestroy {
+export class MacroComponent {
     @ViewChildren(MacroItemComponent) macroItems: QueryList<MacroItemComponent>;
-
-    private macro: Macro;
-    private routeSubscription: Subscription;
-    private hasChanges: boolean = false;
+    private macro$: Observable<Macro>;
```

Comment:
So instead of observable and async in template you would like to have whole object saved in the class and then access it in template without async?